src/state.rs

```diff
@@ -575,8 +644,8 @@ impl DemandProxyState {
             }
         };
         let wp_socket_addr = SocketAddr::new(wp_nw_addr.address, gw_address.hbone_mtls_port);
-        match self
-            .fetch_upstream(&wp_nw_addr.network, wp_socket_addr)
+        match self // THIS IS THE WRONG WORKLOAD
```

Comment:
just need to pass in src, right?

Reply:
ah good catch, forgot about this todo. Yep